src/shm_main.c

```diff
@@ -929,8 +1279,8 @@ sr_shmmain_conn_recover(sr_conn_ctx_t *conn)
     shm_conn = (sr_conn_shm_t *)(conn->ext_shm.addr + main_shm->conns);
     i = 0;
     while (i < main_shm->conn_count) {
-        if (!sr_process_exists(shm_conn[i].pid)) {
-            SR_LOG_WRN("Cleaning up after a non-existent sysrepo client with PID %ld.", (long)shm_conn[i].pid);
+        if (!sr_connection_exists(shm_conn[i].cid)) {
+            SR_LOG_WRN("Cleaning up after a non-existent sysrepo client with CID %ld.", (long)shm_conn[i].cid);
```

Comment:
Please remove the cast here as well, there should be none. Also, the only fully portable way of printing it is probably using `PRIu32` even though it is a bit of a mess, using it everywhere.

Reply:
Done, here and elsewhere.